src/main/java/seedu/address/storage/JsonSerializableAddressBook.java

```
@@ -78,7 +78,9 @@ private void linkPairings(List<PendingPairing> pendingPairings, Map<String, Pers
             Person owner = pending.owner;
             for (JsonAdaptedPerson.PersonIdentity targetIdentity : pending.pairingIdentities) {
                 Person target = identityMap.get(targetIdentity.toLookupKey());
-                if (target == null || target == owner) {
+
+                // The case of (target == owner) is already addressed by Person.addPerson() and the catch clause below
```

Nit: Should we throw a runtime exception if `target == owner` to catch programmer errors (such as if someone modifies `Person.java` in the future to incorrectly allow self-pairings)?